app/controllers/comments_controller.rb

```diff
@@ -1,2 +1,37 @@
 class CommentsController < ApplicationController
+  # The new action is responsible for rendering the form to create a new comment.
+  # It retrieves the corresponding post based on the post_id parameter,
+  # creates a new comment instance, and sets @user to the @current_user.
+  # It's assumed that @current_user is set elsewhere in the application, likely through user authenticatio
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+    @user = @current_user
+  end
+
+  # The create action is responsible for handling the form submission to create a new comment.
+  # It first finds the user based on the user_id parameter,
+  # then finds the corresponding post based on the post_id parameter.
+  # It then initializes a new comment associated with the post and the user,
+  # using the comment_params method to permit only the necessary parameters (in this case, just :text).
+  def create
+    @user = User.find(params[:user_id]) # Get the user from params
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params.merge(user: @user))
```

Comment:
- [ ] Nice work creating the method to add new comment, everything works as it intend to :clap: However, according to the project requirement, comment should be create on the behalf of the user set in the `ApplicationController` (see image). Therefore I suggest you make use of the `current_user` method that you created in your `ApplicationController`. 
![Screenshot from 2023-11-10 10-34-22](https://github.com/Aqurds/coding-thought/assets/91142432/dbd6e047-6a0e-4883-abf7-abb04f7f91ab)
  **HIT:**  Change the line 18 of your code to look like this `@user = current_user`

Reply:
Updated according to suggestion :)